src/main/scala/sangria/relay/Connection.scala

```diff
@@ -32,21 +31,25 @@ object Connection {
     case _ ⇒ false
   }
 
-  def definition[Ctx, Conn[_], Val](
+  def definition[Ctx, Conn[_, _ <: PageInfo], Val](
     name: String,
     nodeType: OutputType[Val],
+    pageInfoType: OutputType[PageInfo] = PageInfo.defaultPageInfoType,
```

Comment:
Related to a previous comment. I think you would need a type parameter for `PageInfo` here. Then you can use it for `pageInfoType` and `connEv`:

```scala
pageInfoType: OutputType[P] = PageInfo.defaultPageInfoType,

/// and later 

implicit connEv: ConnectionLike[Conn, P, Val, Edge[Val]]
```

Reply:
When we use type parameter for `PageInfo` on `Connection.definition`, it would take affect existing codes and tests.  
I think `Connection.definition` is the default connection definition for relay, and `Connection.definitionWithEdge` is for customize it.   
Thus, its name `definitionWithEdge` is may be not exact for it.

Reply:
I see, it's a good point. Maybe it would make sense then to get rid of `pageInfoType` argument in `Connection.definition` and keep it only in `Connection.definitionWithEdge`? I wonder how useful it is to be able to customize the GraphQL type for a default `PageInfo` type in `Connection.definition`.